debug_toolbar/store.py

```diff
@@ -0,0 +1,127 @@
+import json
+from collections import defaultdict, deque
+from typing import Any, Dict, Iterable
+
+from django.core.serializers.json import DjangoJSONEncoder
+from django.utils.encoding import force_str
+from django.utils.module_loading import import_string
+
+from debug_toolbar import settings as dt_settings
+
+
+class DebugToolbarJSONEncoder(DjangoJSONEncoder):
+    def default(self, o: Any) -> Any:
+        try:
+            return super().default(o)
+        except TypeError:
+            return force_str(o)
```

Comment:
I'm wondering if this is a good default for now or if we should rather let an eventual exception bubble up so that panels are forced to only save primitive data types?

Reply:
I think we're going to need this because we can't let an exception bubble up. We're going to need to store a snapshot of a template's context which can include any type.

Reply:
I'm not 100% convinced that silently stringifying is a good idea, but your point makes sense.

Reply:
I removed this with a note in the code for future reference. I decided it's better to have less code and make changes than to plan for any possibility.